Automattic/wp-calypso
# Welcome Tour Next: Base functionality - step repositioning & spotlight

A feature PR that introduces step repositioning and spotlight effect to the welcome tour. Introduces the new functionality behind a URL param for testing.

#### Changes proposed in this Pull Request

Sets the ground for moving/positioning the tour near the element the various steps refer to - so going from a static tour to a more step-by-step guided tour. Main concepts being introduced:

1. Popper.js to move the tour near reference elements on the page.
2. A dark overlay that blocks mouse events from reaching the editor. Focus is trapped inside the tour.
3. A spotlight effect to highlight reference elements behind the overlay. 
4. Hides the functionality behind a search param (`welcome-tour-next`).

#### Media

![Kapture 2021-11-08 at 17 00 04](https://user-images.githubusercontent.com/1705499/140766847-31881e64-63c4-411e-9fdd-5dca39dbcb57.gif)

#### Testing instructions



1. Sandbox a site, checkout branch, and `yarn dev --sync` from `apps/editing-toolkit` to sync sandbox.
2. Go to the editor, append `&welcome-tour-next` to the URL, reload. 
3. Open the sidebar and from there click on "Welcome Guide".
3. Tour should be focused and keyboard navigation enabled.
4. Focus should be on the "Try it out!" button.
5. Stepping through the tour should move steps closer to the element they refer to.
6. Reference elements should be highlighted, as in the demo.
7. Navigating via Tab key should trap the focus within the tour if the tour is expanded (not minified).
8. Minifying the tour should show the editor.
9. Removing `&welcome-tour-next` from the URL and trying again should open the previous tour (no step repositioning, no dark overlay, no spotlight/highlighting of reference elements).

#### TODO

This is a foundation branch. Current and new issues being added to a GH project: https://github.com/Automattic/wp-calypso/milestone/365



Closes #56353 #57056
